inception-project/inception
# #2562 - Project name no longer unique

**What's in the PR**
- Drop unique index on name
- Change hashCode / equals on Project from name to slug
- Automatically add slugs to all project which do not have one yet during startup
- No longer create unique name when importing a project and the name was already taken
- No longer check if project name is already used when setting project name in project details / dashboard
- Adjustments to the remote APIs: the project "name" parameter in the remote API requests and responses is actually the URL slug and the "title" parameter is the project name. In the legacy remote API responses, the slug replaces the name.
- The first line of a project description is now shown as a short description in the project overview page

**How to test manually**
* Start up with external or embedded DB and check if the slugs are auto-generated
* Check if the slugs were actually persisted by hovering with the mouse cursor over the links in the project dashboard
* Try giving two projects the same name manually
* Export a project, then re-import it 

**Automatic testing**

**Documentation**
